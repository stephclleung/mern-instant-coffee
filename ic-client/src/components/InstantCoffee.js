import React from 'react';
import { Link } from 'react-router-dom';
import { Col, Card, CardBody, CardImg, CardText, CardSubtitle, CardTitle } from 'reactstrap';
import bean from '../img/bean.jpg';

const InstantCoffee = ({ id, coffeeName, packageSize, containerSize, price, currency, aroma, acidity }) => {
    return (
        <Col sm="12" className="mb-3">
            <Card style={{ width: 25 + 'rem' }} className="flex-md-row border-dark shadow-sm">
                <CardBody className="align-items-start">
                    <CardTitle ><h4><Link id="coffeeCardTitle" to={`/edit/${id}`}>{coffeeName}</Link></h4></CardTitle>
                    <CardSubtitle>$ {price} {currency} </CardSubtitle>
                    <CardText>
                        <li>Aroma : {renderBeans(aroma)}</li>
                        <li>Acidity : {renderBeans(acidity)}</li>
                        {packageSize > 0 ? (
                            <li>{packageSize} sticks x {containerSize} grams</li>
                        ) : (
                                <li>{containerSize} grams </li>
                            )
                        }
                    </CardText>
                </CardBody>
                <Col md="4" className="my-auto">
                    <CardImg
                        className="img-responsive"
                        height={100 + "%"}
                        src="https://target.scene7.com/is/image/Target/GUEST_600d3360-8929-41a2-b403-7ffc77c7d8de?wid=488&hei=488&fmt=pjpeg"
                        alt="Card image cap"
                    />
                </Col>

            </Card>
        </Col>
    )
}

const renderBeans = (size) => {
    let beans = [];
    for (let b = 0; b < size; b++) {
        beans.push(<img src={bean} key={b} style={{ width: 17 + 'px', height: 18 + 'px' }} alt={"beans"} />)
    }
    return beans
}

export default InstantCoffee;