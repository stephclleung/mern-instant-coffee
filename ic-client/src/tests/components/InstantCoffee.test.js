import InstantCoffee from '../../components/InstantCoffee';
import React from 'react';
import { shallow } from 'enzyme';

test("Should render instant coffee component correctly", () => {
    const wrapper = shallow(<InstantCoffee />)
    expect(wrapper).toMatchSnapshot();
})