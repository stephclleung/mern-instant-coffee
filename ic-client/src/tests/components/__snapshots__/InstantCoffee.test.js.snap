// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render instant coffee component correctly 1`] = `ShallowWrapper {}`;
